test/e2e/managementstate_test.go

```diff
@@ -0,0 +1,141 @@
+	client := framework.MustNewClientset(t, nil)
+
+	defer framework.MustRemoveImageRegistry(t, client)
+
+	framework.MustDeployImageRegistry(t, client, &imageregistryv1.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imageregistryv1.ImageRegistryResourceName,
+		},
+		Spec: imageregistryv1.ImageRegistrySpec{
+			ManagementState: operatorapi.Managed,
+			Replicas:        1,
+		},
+	})
+	framework.MustEnsureImageRegistryIsAvailable(t, client)
+
+	var cr *imageregistryv1.Config
+	var err error
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		cr, err = client.Configs().Get(imageregistryv1.ImageRegistryResourceName, metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cr.Spec.ManagementState = operatorapi.Unmanaged
+
+		cr, err = client.Configs().Update(cr)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO(dmage): wait for the resource to be observed
+
+	err = client.Deployments(imageregistryv1.ImageRegistryOperatorNamespace).Delete(imageregistryv1.ImageRegistryName, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = wait.Poll(1*time.Second, framework.AsyncOperationTimeout, func() (stop bool, err error) {
+		cr, err = client.Configs().Get(imageregistryv1.ImageRegistryResourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		conds := framework.GetImageRegistryConditions(cr)
+		t.Logf("image registry: %s", conds)
+		return conds.Available.IsTrue() && conds.Available.Reason() == "Unmanaged" &&
+			conds.Progressing.IsFalse() && conds.Progressing.Reason() == "Unmanaged" &&
+			conds.Degraded.IsFalse() && conds.Degraded.Reason() == "Unmanaged", nil
+	})
+	if err != nil {
+		framework.DumpImageRegistryResource(t, client)
+		framework.DumpOperatorLogs(t, client)
+		t.Fatal(err)
+	}
+}
+
+func TestManagementStateRemoved(t *testing.T) {
+	client := framework.MustNewClientset(t, nil)
+
+	defer framework.MustRemoveImageRegistry(t, client)
+
+	framework.MustDeployImageRegistry(t, client, &imageregistryv1.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: imageregistryv1.ImageRegistryResourceName,
+		},
+		Spec: imageregistryv1.ImageRegistrySpec{
+			ManagementState: operatorapi.Managed,
+			Replicas:        1,
+		},
+	})
+	framework.MustEnsureImageRegistryIsAvailable(t, client)
+
+	var cr *imageregistryv1.Config
+	var err error
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		cr, err = client.Configs().Get(imageregistryv1.ImageRegistryResourceName, metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cr.Spec.ManagementState = operatorapi.Removed
+
+		cr, err = client.Configs().Update(cr)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO(dmage): wait for the resource to be observed
+
+	err = client.Deployments(imageregistryv1.ImageRegistryOperatorNamespace).Delete(imageregistryv1.ImageRegistryName, &metav1.DeleteOptions{})
```

Comment:
Isn't this Delete automatically happen slightly after we set the Management State to Removed?